generate_outputs.py

```diff
@@ -102,6 +102,20 @@
         log.debug(f"Loaded {len(messages)} messages")
         messages_datasets.append(messages)
 
+    recovery_datasets = []
+    if pipeline_configuration.recovery_csv_urls is None:
+        log.debug("Not loading any recovery datasets (because the pipeline configuration json does not contain the key "
+                  "'RecoveryCSVURLs')")
+    else:
+        log.info("Loading recovery datasets:")
+        for i, recovery_csv_url in enumerate(pipeline_configuration.recovery_csv_urls):
+            raw_recovery_path = f"{raw_data_dir}/{recovery_csv_url.split('/')[-1].split('.')[0]}.json"
```

Comment:
this feels a bit unpleasant, but it also doesn't feel worth a different structural solution.